### PR TITLE
Added `skipWhile` operator

### PR DESCRIPTION
**Pull request by davidwdan**

### CODE REVIEW

File: lib/Rx/Operator/SkipWhileOperator.php [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    private $index;
+
+    /** @var bool */
+    private $lastResult;
+
+    public function __construct(callable  $predicate)
+    {
+        $this->predicate  = $predicate;
+        $this->index      = 0;
+        $this->lastResult = false;
```
Comment (asm89) [inline lib/Rx/Operator/SkipWhileOperator.php]: We should close over the `$index` and `$lastResult` in `__invoke()`. There's nothing stopping a developer from using the same operator instance twice.

**Comment by asm89** [comment]
Can you split into `skipUntil()` and `skipUntilWithIndex()`?

File: lib/Rx/Operator/SkipWhileOperator.php (hunk 1 above)
Comment (asm89) [inline lib/Rx/Operator/SkipWhileOperator.php]: Never mind. Just chatted with @davidwdan on IRC. `lift()` only takes a _factory_ so you always get a new operator. Unless you go full clowntown of course, but then you deserve it? :)

File: lib/Rx/Operator/SkipWhileOperator.php [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    public function __invoke(ObservableInterface $observable, ObserverInterface $observer, SchedulerInterface $scheduler = null)
+    {
+        $callbackObserver = new CallbackObserver(
+            function ($value) use ($observer, $observable) {
+                try {
+
+                    if (!$this->lastResult) {
+                        $result = call_user_func_array($this->predicate, [$value, $observable]);
+
+                        $this->lastResult = !$result;
```
Comment (asm89) [inline lib/Rx/Operator/SkipWhileOperator.php]: Can we rename this to `$this->isSkipping` or something? E.g.

```
if ($this->isSkipping) {
  $continueSkipping = call_user_func...;
  $this->isSkipping = $continueSkipping;
} else {
  $this->onNext($value);
}
```

File: demo/skip/skipWhileWithIndex.php [hunk 3]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+require_once __DIR__ . '/../bootstrap.php';
+
+$observable = Rx\Observable::range(1, 5)
+    ->skipWhileWithIndex(function ($i) {
+        return $i < 3;
+    });
```
Comment (mbonneau) [inline demo/skip/skipWhileWithIndex.php:8]: Can we add the value to this example for clarity?

File: lib/Rx/Observable.php [hunk 4]
```diff
@@ -349,6 +350,40 @@ public function skip($count)
     }
 
     /**
+     * Bypasses elements in an observable sequence as long as a specified condition is true and then returns the
+     * remaining elements. The element's index is used in the logic of the predicate function.
```
Comment (mbonneau) [inline lib/Rx/Observable.php]: This is the non-index version - so the index is not used correct?

File: lib/Rx/Operator/SkipWhileOperator.php [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+class SkipWhileOperator implements OperatorInterface
+{
+    /** @var callable */
+    private $predicate;
+
+    /** @var bool */
+    private $isSkipping;
+
+    public function __construct(callable  $predicate)
```
Comment (mbonneau) [inline lib/Rx/Operator/SkipWhileOperator.php]: nit: 2 spaces between callabel and $predicate

**Pull request merged**